rs/execution_environment/src/scheduler.rs

```diff
@@ -2088,7 +2092,14 @@ fn execute_canisters_on_thread(
         if let Some(es) = &mut canister.execution_state {
             es.last_executed_round = round_id;
         }
-        if !canister.has_input() || rank == 0 {
+        let full_message_execution = match canister.next_execution() {
+            NextExecution::None => true,
+            NextExecution::StartNew => false,
+            // We just finished a full slice of executions.
+            NextExecution::ContinueLong | NextExecution::ContinueInstallCode => true,
```

Comment:
Above, we `break` if `canister.next_execution() == ContinueInstallCode`, i.e. we don't execute anything. It seems incorrect to now claim that such a canister got a full round.